File: backend/app/schemas/message.py
"""消息相关 schema。"""
from pydantic import BaseModel, Field


class NotificationSettingsSchema(BaseModel):
    systemEnabled: bool = True
    securityEnabled: bool = True
    contentEnabled: bool = True


class NotificationSettingsUpdate(BaseModel):
    systemEnabled: bool | None = None
    securityEnabled: bool | None = None
    contentEnabled: bool | None = None


class MessageListResult(BaseModel):
    items: list[dict]
    unreadCount: int

File: src/shared/types/message.ts
export type MessageType = 'system' | 'comment' | 'reminder' | 'update'
export type MessageCategory = 'system' | 'security' | 'content'

export interface MessageItem {
  id: string
  title: string
  summary: string
  content: string[]
  time: string
  createdAt: string
  type: MessageType
  category: MessageCategory
  source: string
  tag: string
  unread: boolean
  primaryAction?: string
  secondaryAction?: string
}

export interface NotificationSettings {
  systemEnabled: boolean
  securityEnabled: boolean
  contentEnabled: boolean
  emailEnabled: boolean
}

export interface MessageListResult {
  items: MessageItem[]
  unreadCount: number
}
